tests/core/io/test_stream_peer_tcp.h

```diff
@@ -0,0 +1,303 @@
+	_is_open = true;
+	return OK;
+}
+void MockNetSocket::close() {
+	_is_open = false;
+}
+Error MockNetSocket::bind(IPAddress p_addr, uint16_t p_port) {
+	host_ip = p_addr;
+	host_port = p_port;
+	return OK;
+}
+Error MockNetSocket::listen(int p_max_pending) {
+	return OK;
+}
+Error MockNetSocket::connect_to_host(IPAddress p_addr, uint16_t p_port) {
+	dest_ip = p_addr;
+	dest_port = p_port;
+	return OK;
+}
+Error MockNetSocket::poll(PollType p_type, int timeout) const {
+	return OK;
+}
+Error MockNetSocket::recv(uint8_t *p_buffer, int p_len, int &r_read) {
+	// Receives one byte of _recv_data on each invocation.
+	p_buffer[0] = _recv_data[0];
+	_recv_data += 1;
+	r_read = 1;
+	return OK;
+}
+Error MockNetSocket::recvfrom(uint8_t *p_buffer, int p_len, int &r_read, IPAddress &r_ip, uint16_t &r_port, bool p_peek) {
+	return OK;
+}
+Error MockNetSocket::send(const uint8_t *p_buffer, int p_len, int &r_sent) {
+	// Sends one byte to _sent_data on each invocation.
+	_sent_data[0] = p_buffer[0];
+	_sent_data += 1;
+	r_sent = 1;
+	return OK;
+}
+Error MockNetSocket::sendto(const uint8_t *p_buffer, int p_len, int &r_sent, IPAddress p_ip, uint16_t p_port) {
+	return OK;
+}
+Ref<NetSocket> MockNetSocket::accept(IPAddress &r_ip, uint16_t &r_port) {
+	return this;
+}
+
+bool MockNetSocket::is_open() const {
+	return _is_open;
+}
+int MockNetSocket::get_available_bytes() const {
+	return _available_bytes;
+}
+Error MockNetSocket::get_socket_address(IPAddress *r_ip, uint16_t *r_port) const {
+	return OK;
+}
+
+Error MockNetSocket::set_broadcasting_enabled(bool p_enabled) {
+	return OK;
+}
+void MockNetSocket::set_blocking_enabled(bool p_enabled) {
+	blocking_enabled = p_enabled;
+}
+void MockNetSocket::set_ipv6_only_enabled(bool p_enabled) {}
+void MockNetSocket::set_tcp_no_delay_enabled(bool p_enabled) {}
+void MockNetSocket::set_reuse_address_enabled(bool p_enabled) {}
+Error MockNetSocket::join_multicast_group(const IPAddress &p_multi_address, const String &p_if_name) {
+	return OK;
+}
+Error MockNetSocket::leave_multicast_group(const IPAddress &p_multi_address, const String &p_if_name) {
+	return OK;
+}
+
+MockNetSocket::MockNetSocket() {}
+MockNetSocket::~MockNetSocket() {}
+
+void convert_to_string(uint8_t *in_data, char *out_str, const int str_len) {
+	for (int i = 0; i < str_len; ++i) {
+		out_str[i] = static_cast<char>(in_data[i]);
+	}
+}
+
+TEST_CASE("[StreamPeerTCP] basics") {
+	Ref<MockNetSocket> ns = memnew(MockNetSocket);
+	Ref<StreamPeerTCP> spt = memnew(StreamPeerTCP);
+	IPAddress peer_ip = IPAddress("127.0.1.1");
+	int peer_port = 5678;
+	spt->accept_socket(ns, peer_ip, peer_port);
+	REQUIRE(ns->blocking_enabled == false);
+
+	IPAddress bind_ip = IPAddress("127.0.0.1");
+
+	SUBCASE("invalid port numbers returns an Error") {
+		Error negative_ret = spt->bind(-901, bind_ip);
+		REQUIRE(negative_ret != OK);
+
+		Error too_high_ret = spt->bind(70000, bind_ip);
+		REQUIRE(too_high_ret != OK);
```

Comment:
This test is printing some errors when is run. You can use `ERR_PRINT_OFF/ERR_PRINT_ON` to prevent it

```suggestion
		ERR_PRINT_OFF;
		Error negative_ret = spt->bind(-901, bind_ip);
		REQUIRE(negative_ret != OK);

		Error too_high_ret = spt->bind(70000, bind_ip);
		REQUIRE(too_high_ret != OK);
		ERR_PRINT_ON;
```